common/gera/map_test.go

```diff
@@ -0,0 +1,626 @@
+			Expect(ok).To(BeTrue())
+			Expect(value).To(Equal("test"))
+		})
+
+		It("should correctly return a copy of the whole structure", func() {
+			copy := stringMap.Copy()
+			Expect(copy).NotTo(BeNil())
+			Expect(copy == stringMap).NotTo(BeTrue())
+			Expect(copy).To(Equal(stringMap))
+			Expect(copy.Raw()).To(Equal(stringMap.Raw()))
+			Expect(copy.Len()).To(Equal(stringMap.Len()))
+			copy.Set("detector", "test")
+			Expect(copy).NotTo(Equal(stringMap))
+			Expect(copy.Raw()).NotTo(Equal(stringMap.Raw()))
+			Expect(copy.Len()).To(Equal(stringMap.Len()))
+			copy.Set("detector2", "test2")
+			Expect(copy.Len()).To(Equal(stringMap.Len() + 1))
+		})
+
+		It("should correctly return a copy of the underlying map", func() {
+			rawCopy := stringMap.RawCopy()
+			Expect(rawCopy).NotTo(BeNil())
+			Expect(len(rawCopy)).To(Equal(6))
+			rawCopy["detector"] = "test"
+			value, ok := stringMap.Get("detector")
+			Expect(ok).To(BeTrue())
+			Expect(value).To(Equal(""))
+		})
+	})
+	When("wrapping and unwrapping maps", func() {
+
+		defaults1 := MakeMap[string, string]()
+		Expect(defaults1).NotTo(BeNil())
+
+		defaults1 = defaults1.WithUnmarshalYAML(customUnmarshalYAML)
+		err = yaml.Unmarshal([]byte(testPayloadDefaultsYAML1), defaults1)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(defaults1).NotTo(BeNil())
+
+		defaults2 := MakeMap[string, string]()
+		Expect(defaults2).NotTo(BeNil())
+
+		err = yaml.Unmarshal([]byte(testPayloadDefaultsYAML2), defaults2)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(defaults2).NotTo(BeNil())
+
+		vars1 := MakeMap[string, string]()
+		Expect(vars1).NotTo(BeNil())
+
+		err = yaml.Unmarshal([]byte(testPayloadVarsYAML1), vars1)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(vars1).NotTo(BeNil())
+
+		vars2 := MakeMap[string, string]()
+		Expect(vars2).NotTo(BeNil())
+
+		err = yaml.Unmarshal([]byte(testPayloadVarsYAML2), vars2)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(vars2).NotTo(BeNil())
+
+		userVars1 := MakeMap[string, string]()
+		Expect(userVars1).NotTo(BeNil())
+
+		err = yaml.Unmarshal([]byte(testPayloadUserVarsYAML1), userVars1)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(userVars1).NotTo(BeNil())
+
+		var wrappedDefaults, wrappedVars, wrappedUserVars, wrappedAll Map[string, string]
+
+		It("should wrap correctly", func() {
+			wrappedDefaults = defaults2.Wrap(defaults1)
+			Expect(wrappedDefaults).NotTo(BeNil())
+			Expect(wrappedDefaults).To(Equal(defaults2))
+
+			wrappedVars = vars2.Wrap(vars1)
+			Expect(wrappedVars).NotTo(BeNil())
+			Expect(wrappedVars).To(Equal(vars2))
+
+			wrappedUserVars = userVars1
+		})
+
+		It("should unwrap correctly", func() {
+			unwrapped := vars2.Unwrap()
+			Expect(unwrapped).NotTo(BeNil())
+			Expect(unwrapped).To(Equal(vars1))
+
+			wrappedVars = vars2.Wrap(vars1)
+			Expect(wrappedVars).NotTo(BeNil())
+		})
+
+		var flattenedDefaults, flattenedVars, flattenedUserVars, flattenedAll map[string]string
+
+		It("should flatten correctly", func() {
```

Comment:
By `Flatten()`, do you mean to remove the nesting by representing the whole path to a value in the key? If so, could we perhaps test this? I don't quite understand what do we test now.

Reply:
No, flattening doesn't modify keys. It just merges a hierarchy of wrapped maps into a single `map[K]V` while respecting the override order.

Reply:
thanks, clear